packages/main/src/plugin/kubernetes/contexts-states.ts

```diff
@@ -135,7 +201,83 @@ export class ContextsStates {
     update(val);
   }
 
-  async dispose(name: string): Promise<void> {
+  dispatch(options: DispatchOptions): void {
+    if (options.checkingState) {
+      this.dispatchCheckingState(this.getContextsCheckingState());
+    }
+    if (options.contextsGeneralState) {
+      this.dispatchGeneralState(this.getContextsGeneralState());
+    }
+    if (options.currentContextGeneralState) {
+      this.dispatchCurrentContextGeneralState(this.getCurrentContextGeneralState(options.currentContext));
+    }
+    Object.keys(options.resources).forEach(res => {
+      const resname = res as ResourceName;
+      if (options.resources[resname]) {
+        this.dispatchCurrentContextResource(resname, this.getContextResources(options.currentContext, resname));
+      }
+    });
+  }
+
+  public dispatchCheckingState(val: Map<string, CheckingState>): void {
+    this.apiSender.send(`kubernetes-contexts-checking-state-update`, val);
+  }
+
+  public dispatchGeneralState(val: Map<string, ContextGeneralState>): void {
+    this.dispatchContextsGeneralStateTimer = this.dispatchDebounce(
+      `kubernetes-contexts-general-state-update`,
+      val,
+      this.dispatchContextsGeneralStateTimer,
+      dispatchTimeout,
+    );
+  }
+
+  public dispatchCurrentContextGeneralState(val: ContextGeneralState): void {
+    this.dispatchCurrentContextGeneralStateTimer = this.dispatchDebounce(
+      `kubernetes-current-context-general-state-update`,
+      val,
+      this.dispatchCurrentContextGeneralStateTimer,
+      dispatchTimeout,
+    );
+  }
+
+  public dispatchCurrentContextResource(resname: ResourceName, val: KubernetesObject[]): void {
+    this.dispatchCurrentContextResourceTimers.set(
+      resname,
+      this.dispatchDebounce(
+        `kubernetes-current-context-${resname}-update`,
+        val,
+        this.dispatchCurrentContextResourceTimers.get(resname),
+        dispatchTimeout,
+      ),
+    );
+  }
+
+  private dispatchDebounce(
+    eventName: string,
+    value: Map<string, ContextGeneralState> | ContextGeneralState | KubernetesObject[] | Map<string, boolean>,
+    timer: NodeJS.Timeout | undefined,
+    timeout: number,
+  ): NodeJS.Timeout | undefined {
+    clearTimeout(timer);
+    if (this.disposed) {
+      return undefined;
+    }
+    return setTimeout(() => {
+      this.apiSender.send(eventName, value);
+    }, timeout);
+  }
+
+  dispose(): void {
+    this.disposed = true;
+    clearTimeout(this.dispatchContextsGeneralStateTimer);
+    clearTimeout(this.dispatchCurrentContextGeneralStateTimer);
+    for (const timer of this.dispatchCurrentContextResourceTimers.values()) {
+      clearTimeout(timer);
+    }
+  }
+
+  async disposeContext(name: string): Promise<void> {
```

Comment:
`disposeContext` is not called when `dispose()` is called ?

Reply:
`disposeContext` is probably badly named, as it is not disposing system resources, as `dispose` does, but only _deletes_ an entry from the map.

I'm pushing a new commit to rename disposeContext and ContextInformers.dispose with a better name IMO